chainer/gradient_check.py

```diff
@@ -48,18 +44,22 @@ def numerical_grad(f, inputs, grad_outputs, eps=1e-3):
     else:
         xp = numpy
     grads = tuple(xp.zeros_like(x) for x in inputs)
-    for x, gx in zip(inputs, grads):
+
+    prev_mode = function.Function.type_check_enable
+    function.Function.type_check_enable = False
+    for x, gx in six.moves.zip(inputs, grads):
         for i in numpy.ndindex(x.shape):
             orig = x[i].copy()  # hold original value
             x[i] = orig + eps
-            ys1 = _copy_arrays(f())
+            ys1 = [xp.copy(j) for j in f()]
             x[i] = orig - eps
-            ys2 = _copy_arrays(f())
+            ys2 = [xp.copy(j) for j in f()]
             x[i] = orig
-            for y1, y2, gy in zip(ys1, ys2, grad_outputs):
+            for y1, y2, gy in six.moves.zip(ys1, ys2, grad_outputs):
                 if gy is not None:
                     dot = ((y1 - y2) * gy).sum()
                     gx[i] += dot / (2 * eps)
+    function.Function.type_check_enable = prev_mode
```

Comment:
If an exception is raised before this statement, `function.Function.type_check_enable` is not restored.